### PR TITLE
Create API functions to create and get telegraf config

### LINKED ISSUE #1461: Add the ability to download a telegraf config file from the UI in toml format
As a new user
with admin permissions or higher
when i have configured my streaming sources 
then i would like the ability to download a telegraf toml file with the settings

### PR DESCRIPTION
**Pull request by ebb-tide**
Closes #1461 

Added API call functions using the generated client for creating and getting telegraf configs. 
Updated cur_swagger to reflect that requests with header {Content: 'application/json'} returns toml file instead of json representation of telegraf. 
Regenerated client. 
Created a button on the ConfigureDataSourceSwitcher component that creates a dummy telegraf config and then downloads the toml config file.

### CODE REVIEW

File: http/cur_swagger.yml [hunk 1]
```diff
@@ -144,8 +144,9 @@ paths:
           description: telegraf config details
           content:
             application/json:
+              example: "[agent]\ninterval = \"10s\""
```
Comment (ischolten) [inline http/cur_swagger.yml]: noice

File: http/cur_swagger.yml [hunk 2]
```diff
@@ -144,8 +144,9 @@ paths:
...
               schema:
-                $ref: "#/components/schemas/Telegraf"
```
Comment (ischolten) [inline http/cur_swagger.yml]: does the response come back as a string that is in the shape of the Telegraf Schema? Im just curious where this (`$ref: "#/components/schemas/Telegraf"`) would go

File: ui/src/api/api.ts [hunk 3]
```diff
@@ -10648,7 +10648,7 @@ export const TelegrafsApiFp = function(configuration?: Configuration) {
          * @param {*} [options] Override http request option.
          * @throws {RequiredError}
          */
-        telegrafsTelegrafIDGet(telegrafID: string, options?: any): (axios?: AxiosInstance, basePath?: string) => AxiosPromise<Telegraf> {
+        telegrafsTelegrafIDGet(telegrafID: string, options?: any): (axios?: AxiosInstance, basePath?: string) => AxiosPromise<string> {
```
Comment (ischolten) [inline ui/src/api/api.ts]: oh cool is this the generated stuff?

File: ui/src/onboarding/apis/index.ts [hunk 4]
```diff
@@ -16,6 +19,35 @@ export const getSetupStatus = async (url: string): Promise<boolean> => {
...
+}
+
+export const createTelegrafConfig = async (): Promise<Telegraf> => {
+  const telegrafRequest: TelegrafRequest = {
+    name: 'testName',
+    agent: {interval: 1},
+    plugins: [
+      {
+        name: TelegrafRequestPlugins.NameEnum.Cpu,
+        type: TelegrafRequestPlugins.TypeEnum.Input,
```
Comment (ischolten) [inline ui/src/onboarding/apis/index.ts]: lovin the enums

File: http/cur_swagger.yml (hunk 2 above)
Comment (ischolten) [reply http/cur_swagger.yml]: gotcha sounds good!

Comment (ebb-tide) [reply http/cur_swagger.yml]: I'm actually reverting this due to further discussion: https://github.com/influxdata/platform/issues/1587

File: ui/src/onboarding/components/ConfigureDataSourceSwitcher.tsx [hunk 5]
```diff
@@ -13,7 +20,18 @@ class ConfigureDataSourceSwitcher extends PureComponent<Props> {
...
+      <div>
+        {dataSource}
+        <Button text="Click to Download Config" onClick={this.handleDownload} />
+      </div>
+    )
+  }
+
+  private handleDownload = async () => {
+    const telegraf = await createTelegrafConfig()
+    const config = await getTelegrafConfigTOML(telegraf.id)
```
[the remark this answers is not in the document]
Comment (ebb-tide) [reply ui/src/onboarding/components/ConfigureDataSourceSwitcher.tsx]: 👍

**Pull request merged**